addons/website_mail/controllers/main.py

```diff
@@ -97,25 +105,56 @@ def is_follower(self, model, res_id, **post):
             ]))
         return values
 
-    @http.route(['/website_mail/post/json'], type='json', auth='public', website=True)
-    def chatter_json(self, res_model='', res_id=None, message='', **kw):
-        try:
-            msg = _message_post_helper(res_model, int(res_id), message, **kw)
-        except Exception:
-            return False
-        data = {
-            'id': msg.id,
-            'body': msg.body,
-            'date': msg.date,
-            'author': msg.author_id.name,
-            'image_url': '/mail/%s/%s/avatar/%s' % (msg.model, msg.res_id, msg.author_id.id)
-        }
-        return data
-
     @http.route(['/website_mail/post/post'], type='http', methods=['POST'], auth='public', website=True)
-    def chatter_post(self, res_model='', res_id=None, message='', redirect=None, **kw):
+    def chatter_post(self, res_model, res_id, message, **kw):
         url = request.httprequest.referrer
         if message:
-            message = _message_post_helper(res_model, int(res_id), message, **kw)
-            url = url + "#message-%s" % (message.id,)
+            self._website_message_post(res_model, res_id, message, **kw)
+            url = url + "#discussion"
         return request.redirect(url)
+
+    def _website_message_post(self, res_model, res_id, message, **kw):
```

Comment:
It is a hook, to create the rating in website_rating. Otherwise, super will return the url, and don't have the message_id to associate the rating with.
Maybe `_message_post_helper` should be a controller method. What do you think ?